swift-sdk/Internal/AuthManager.swift

```diff
@@ -107,14 +107,22 @@ class AuthManager: IterableAuthManagerProtocol {
     private func queueAuthTokenExpirationRefresh(_ authToken: String?) {
         ITBInfo()
         
+        clearRefreshTimer()
+        
         guard let authToken = authToken, let expirationDate = AuthManager.decodeExpirationDateFromAuthToken(authToken) else {
+            /// schedule a default timer of 10 seconds if we fall into this case
+            scheduleAuthTokenRefreshTimer(10)
+            
             return
         }
         
         let timeIntervalToRefresh = TimeInterval(expirationDate) - dateProvider.currentDate.timeIntervalSince1970 - expirationRefreshPeriod
         
-        clearRefreshTimer()
-        expirationRefreshTimer = Timer.scheduledTimer(withTimeInterval: timeIntervalToRefresh, repeats: false) { [weak self] _ in
+        scheduleAuthTokenRefreshTimer(timeIntervalToRefresh)
+    }
+    
+    private func scheduleAuthTokenRefreshTimer(_ interval: TimeInterval) {
```

Comment:
Method extracted looks helpful 👍🏼. We might have to call it from another few places too.